pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -336,5 +336,12 @@ func validateRolloutStrategy(rolloutStrategy placementv1beta1.RolloutStrategy) e
 		}
 	}
 
+	// server-side apply strategy type is only valid for server-side apply strategy type
+	if rolloutStrategy.ApplyStrategy != nil {
+		if rolloutStrategy.ApplyStrategy.Type != placementv1beta1.ApplyStrategyTypeServerSideApply && rolloutStrategy.ApplyStrategy.ServerSideApplyConfig != nil {
+			allErr = append(allErr, fmt.Errorf("serverSideApplyConfig is only valid for server-side apply strategy type"))
```

Comment:
```suggestion
			allErr = append(allErr, errors.New("serverSideApplyConfig is only valid for ServerSideApply strategy type"))
```
you don't need the format here, so the errors.New is good enough.